### PR TITLE
POC  - track intently modal actions

### PR DESCRIPTION
**Pull request by EdPutans**

### CODE REVIEW

**Review by Luciam91 — APPROVED** [review]

**Review by dbacall — APPROVED** [review]

File: src/hooks/useIntently.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      )
+
+      Array.prototype.forEach.call(locatedIntentlyScript, (node: any) => {
+        node.parentNode.removeChild(node)
+
+        log('useRemoveIntently: successfully removed intently overlay')
+
+        clearInterval(runningInterval)
+      })
+    }, 100)
```
Comment (EdPutans) [inline src/hooks/useIntently.ts:95]: Note to self: maybe replace 250 with 100 as here

File: src/hooks/useIntently.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+ */
+
+function useTrackIntentlyModal() {
+  const [isVisible, setIsVisible] = useState<boolean>(false)
+  const { trackEvent } = useMixpanel()
+  const { log, error } = useLogging()
+
+  useEffect(() => {
+    const id = setInterval(() => {
+      const intentlyInnerOverlay = document.querySelector(
```
Comment (Luciam91) [inline src/hooks/useIntently.ts:18]: Unfortunately this isn't enough to say this is visible, we need to wait until the parent property is set to `display:block`

Comment (EdPutans) [reply src/hooks/useIntently.ts]: guessing this is not an option?
https://caniuse.com/?search=checkvisibility

File: src/hooks/useIntently.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+      if (!intentlyInnerOverlay) {
+        log(
+          'useTrackIntentlyModal: Could not locate intently overlay inner content, not tracking performance.'
+        )
+        return
+      }
+
+      log(
+        'useTrackIntentlyModal: Located Intently modal. Releasing the Kraken.'
```
Comment (Luciam91) [inline src/hooks/useIntently.ts:46]: Nit. Can we remove the `Releasing the Kraken`

In theory these logs could be shown to a customer in the browser console

Comment (EdPutans) [reply src/hooks/useIntently.ts:46]: ...Nothing wrong with devs having a little fun... 😓  

also, not a theory, they are visible atm

**Comment by EdPutans** [comment]
TODO BEFORE MERGE:

make sure all "AddPageTriggers" are replaced with "SetTriggers"

**Review by Luciam91 — APPROVED** [review]

**Pull request merged**